strawberry/field.py

```diff
@@ -234,17 +287,22 @@ def copy_with(
             default_factory=self.default_factory,  # type: ignore[misc]
             deprecation_reason=self.deprecation_reason,
         )
+        field.type = new_type
+        return field
 
     def get_result(
-        self, source: Any, info: Info, args: List[Any], kwargs: Dict[str, Any]
+        self, source: Any, info: Info, arguments: Dict[str, Any]
     ) -> Union[Awaitable[Any], Any]:
         """
         Calls the resolver defined for the StrawberryField. If the field doesn't have a
         resolver defined we default to using getattr on `source`.
         """
 
         if self.base_resolver:
-            return self.base_resolver(*args, **kwargs)
+            args = []
+            if self.base_resolver.has_self_arg:
+                args = [source]
+            return self.base_resolver(*args, **arguments)
```

Comment:
Do you know how this works with async resolvers?

Reply:
It should be fine? I haven't changed it's behaviour from before, just moved the optional source arg logic here.